packages/elements/src/number-field/__test__/number-field.test.js

```diff
@@ -306,6 +306,101 @@ describe('number-field/NumberField', function () {
       await elementUpdated(el);
       expect(el.value).to.equal('0', 'value should not be less then zero');
     });
+    it('Arrow up should fire up-click and value-changed', async function () {
+      let upClickedCount = 0;
+      let valueChangedCount = 0;
+      el.addEventListener('up-click', () => {
+        upClickedCount += 1;
+      });
+      el.addEventListener('value-changed', () => {
+        valueChangedCount += 1;
+      });
+      el.inputElement.dispatchEvent(new KeyboardEvent('keydown', { key: 'ArrowUp' }));
+      await elementUpdated(el);
+      expect(el.value).to.equal('1');
+      expect(upClickedCount).to.equal(1);
+      expect(valueChangedCount).to.equal(1);
+    });
+    it('Arrow down should fire down-click and value-changed', async function () {
+      let downClickedCount = 0;
+      let valueChangedCount = 0;
+      el.addEventListener('down-click', () => {
+        downClickedCount += 1;
+      });
+      el.addEventListener('value-changed', () => {
+        valueChangedCount += 1;
+      });
+      el.inputElement.dispatchEvent(new KeyboardEvent('keydown', { key: 'ArrowDown' }));
+      await elementUpdated(el);
+      expect(el.value).to.equal('-1');
+      expect(downClickedCount).to.equal(1);
+      expect(valueChangedCount).to.equal(1);
+    });
+    it('Tapping spinner up should fire up-click and value-changed', async function () {
+      let upClickedCount = 0;
+      let valueChangedCount = 0;
+      el.addEventListener('up-click', () => {
+        upClickedCount += 1;
+      });
+      el.addEventListener('value-changed', () => {
+        valueChangedCount += 1;
+      });
+      setTimeout(() => dispatchTapEvent(spinnerUpEl));
+      await oneEvent(spinnerUpEl, 'tap');
+      await elementUpdated(el);
+      expect(el.value).to.equal('1');
+      expect(upClickedCount).to.equal(1);
+      expect(valueChangedCount).to.equal(1);
+    });
+    it('Tapping spinner down should fire down-click and value-changed', async function () {
+      let downClickedCount = 0;
+      let valueChangedCount = 0;
+      el.addEventListener('down-click', () => {
+        downClickedCount += 1;
+      });
+      el.addEventListener('value-changed', () => {
+        valueChangedCount += 1;
+      });
+      setTimeout(() => dispatchTapEvent(spinnerDownEl));
+      await oneEvent(spinnerDownEl, 'tap');
+      await elementUpdated(el);
+      expect(el.value).to.equal('-1');
+      expect(downClickedCount).to.equal(1);
+      expect(valueChangedCount).to.equal(1);
+    });
+    it('Should stop view-changed event when set prevent default in up-click', async function () {
```

Comment:
```suggestion
    it('Should stop value-changed event when set prevent default in up-click', async function () {
```

Reply:
or how about `value-changed event should not fires when set prevent default to up-click`?

Reply:
Updated by following your rephase suggestion.